### DISCUSSION TITLE
`tokio_stream::StreamExt` vs. `futures::StreamExt` in chat example

### DISCUSSION THREAD

**Discussion by atsmtat** [tokio-rs/tokio#4855] [General] [answered]
Hello, I was following up `examples/chat.rs` example. I noticed that it uses `StreamExt` from `tokio_stream`, but for `SinkExt`, it uses `futures` crate, which I found a bit confusing. Is that purely a relic from the time when `StreamExt` was not part of `futures`? Or is one preferable over the other? I can submit a PR fixing it, unless there's a catch.

**Comment by Darksonn** [comment, marked as answer]
Both `futures::stream::Stream` and `tokio_stream::Stream` are the same trait, so it makes no difference. 

(They're both re-exports of `futures_core::stream::Stream`.)

**Reply by atsmtat** [reply]
Hmm... isn't `StreamExt` (trait which extends`Stream`) in [futures](https://docs.rs/futures/0.3.21/futures/stream/trait.StreamExt.html) different from one in [tokio_stream](https://docs.rs/tokio-stream/0.1.9/tokio_stream/trait.StreamExt.html)? I guess it doesn't matter for the chat example as the method it uses is defined by both. 

I just noticed a comment on `tokio_stream` page, which explicitly warns about these two different implementations of `StreamExt`. Never mind then. Thanks for looking!

**Reply by atsmtat** [reply]
If you think they should be merged into single `StreamExt`, just like `Stream`, then let me know. I'll be happy to pursue this further.

**Reply by Darksonn** [reply]
Right, the `StreamExt` traits are different. Anyway, just leave it as it is.
